years/2020/01/solutions.ts

```diff
@@ -0,0 +1,33 @@
+export const calculate = (input: number[], num: number) => {
+  const match = combinations(input, num).find((combination: number[]) => {
+    return combination.reduce((calculation: number, currentNumber: number) => calculation + currentNumber, 0) === 2020
+  })
+
+  if (!match) {
+    throw new Error(`Sorry, couldn't find a match`)
+  }
+
+  return match.reduce((a: number, b: number) => a * b, 1)
+}
+
+function combinations(set: number[], k: number): number[][] {
+  if (k > set.length) return []
+  if (k == set.length) return [set]
+
+  if (k == 1) {
+    let con: number[][] = []
+    set.forEach((_, index: number) => {
+      con.push([set[index]])
+    })
+    return con
+  }
+
+  let con: number[][] = []
+  for (let i = 0; i < set.length - k + 1; i++) {
+    let head = set.slice(i, i + 1)
```

Comment:
Does this achieve the same thing?

```suggestion
    let head = set[i]
```

Reply:
Nope - this will return the value in the array at that index. We actually want to slice the array:

```javascript
> set = [1, 2, 3]
[ 1, 2, 3 ]
> set.slice(0, 2 + 1)
[ 1, 2, 3 ]
> set[2]
3
```

Reply:
Oh right, yeah 🙈